src/map/skill.cpp

```diff
@@ -9767,43 +9765,45 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 	case SA_SPELLBREAKER:
 		{
 			int sp;
-			if(tsc && tsc->getSCE(SC_MAGICROD)) {
-				sp = skill_get_sp(skill_id,skill_lv);
-				sp = sp * tsc->getSCE(SC_MAGICROD)->val2 / 100;
-				if(sp < 1) sp = 1;
-				status_heal(bl,0,sp,2);
-				status_percent_damage(bl, src, 0, -20, false); //20% max SP damage.
-			} else {
-				struct unit_data *ud = unit_bl2ud(bl);
-				int bl_skill_id=0,bl_skill_lv=0,hp = 0;
+			if (dstsd && tsc && tsc->getSCE(SC_MAGICROD)) {
+				// If target enemy player has Magic Rod, then 20% of your SP is transferred to that player
+				sp = status_percent_damage(bl, src, 0, -20, false);
+				status_heal(bl, 0, sp, 2);
+			}
+			else {
+				struct unit_data* ud = unit_bl2ud(bl);
+				int bl_skill_id = 0, bl_skill_lv = 0, hp = 0;
 				if (!ud || ud->skilltimer == INVALID_TIMER)
 					break; //Nothing to cancel.
 				bl_skill_id = ud->skill_id;
 				bl_skill_lv = ud->skill_lv;
```

Comment:
GitHub won't let me suggest the removal of the two lines above. You could also define the `int hp = 0;` here after the break!